underscore.js

```diff
@@ -919,6 +919,31 @@
 
   _.restArgs = restArgs;
 
+  // Returns non-deterministic map computation of list of functions applied to combinations of
+  // arguments from given lists of possible arguments. An implementation of lists as applicative functors.
+  _.nMap = function(funcs) {
+    var self = this;
+    var args = Array.prototype.slice.call(arguments, 1);
+
+    if (args.length === 1) {
+      return _.map(funcs, function(func) {
+        return _.map(args[0], func);
+      }).reduce(function(agg, result) {
```

Comment:
Keep in mind the return value for `_.map` is an array. This `reduce` call here relies on `Array.prototype.reduce` which isn't present in pre-es5 environs. Try using `_.chain`.